### PR TITLE
CRM-19016 Order mail processing by contact type to let individuals take precedence

### PR DESCRIPTION
**Pull request by niels-heinemann**
When sending mass mailings some groups may contain organizations and individuals. If a pair of them share the same mail address (let's say »office@school.org«) and deduping is engaged, either the organization or the person will receive the mailing by random. But those emails may differ by token substitutions and thus greeting phrases. The organization would get an impersonal opening like »dear sirs and madams« while the person would get a personal salutation like "Dear Jane Doe". A team like a school's teaching stuff working with a shared email account would know to whom the mail concerns if we could guarantee that the persons get served first.

See [Improvement CRM-19016](https://issues.civicrm.org/jira/browse/CRM-19016?filter=-2&jql=reporter%20%3D%20currentUser%28%29%20ORDER%20BY%20key%20DESC%2C%20createdDate%20DESC)

Although a perfect solution would include a better handling of dýnamicly changing contact type names, I just added a simple order by clause because performance tests with ~200k contacts increased consumed time by factor 10. And "Individual" comes before "Organization" in standard installations.
See line comment.

### CODE REVIEW

File: CRM/Mailing/BAO/MailingJob.php [hunk 1]
```diff
@@ -520,7 +520,8 @@ public function deliver(&$mailer, $testParams = NULL) {
                         AND     $edTable.id IS null
                         AND     $ebTable.id IS null
                         AND    contact_a.is_opt_out = 0 
-                        $aclWhere";
+                        $aclWhere
+                    ORDER BY    $contactTable.contact_type";
```
Comment (nielosz) [inline CRM/Mailing/BAO/MailingJob.php]: A solution like 
`
ORDER BY IF($contact.contact_type = 'Individual', 0, 1)
`
would fit the needs more exactly. But it increases time consumption by factor 10.

Comment (jitendrapurohit) [inline CRM/Mailing/BAO/MailingJob.php]: Jenkins test failure seems related. Should be fixed by using the alias(contact_a) instead of `$contactTable`.

**Comment by nielosz** [comment]
Sorry! Used 4.7.6 source code any copied by accident.

**Comment by davejenx** [comment]
@nielosz We're reviewing PRs at the Edale code sprint this week. The relevant part of CRM/Mailing/BAO/MailingJob.php appears to have changed since you submitted your PR, presumably this is why checks are now failing. Are you able to amend your changes to work with the current master code? Apologies on behalf of the community for the delay in reviewing and consequent inconvenience.

**Comment by nielosz** [comment]
Messed PR. Will clean tomorrow....

**Comment by nielosz** [comment]
Updated branch, but PR seems not recognizing it?!

**Comment by nielosz** [comment]
Closed in favour of PR 9236

**Pull request closed without merging**